src/data/polynomial/ring_division.lean

```diff
@@ -144,14 +144,14 @@ have p = X - C (- p.coeff 0),
 this.symm ▸ prime_X_sub_C
 
 theorem irreducible_X_sub_C (r : R) : irreducible (X - C r) :=
-irreducible_of_prime prime_X_sub_C
+prime.irreducible prime_X_sub_C
```

Comment:
Does that not work? (untested) Same for `polynomial.irreducible_X` and `padic_int.irreducible_p`.
```suggestion
prime_X_sub_C.irreducible
```
Also, do you want to make the explicitness of `r` consistent between `polynomial.prime_X_sub_C` and `polynomial.irreducible_X_sub_C`?

Reply:
This one works only if `r` is explicit. Otherwise, the elaborator has to make too many guesses to figure out what the `.irreducible` dot notation is applied to. So `prime_X.irreducible` below would not work.